File: solution/마리오게임/마리오게임_solution.java
import java.util.Scanner;

public class Main {
	int N;//버섯 수
	int P[] = new int[150000 + 10];//버섯 값

	public void inputData()  {
		Scanner sc = new Scanner(System.in);
		
		N = sc.nextInt();
		for (int i = 0; i < N; i++) {
			P[i] = sc.nextInt();
		}
	}
	
	public int solve(){
		int sum = P[0];
		for(int i = 1; i < N; i++){
			if(P[i - 1] < P[i]) sum = sum - P[i - 1] + P[i];
		}
		return sum;
	}

	public static void main(String[] args){
		Main m = new Main();
		int ans = -1;

		m.inputData();				//	입력 함수

		//	코드를 작성하세요
		ans = m.solve();
	
		System.out.println(ans);	//	정답 출력
	}
}

File: solution/마리오게임/마리오게임_solution.c
#include <stdio.h>

int N;//버섯 수
int P[150000 + 10];//버섯 값

void Input_Data(void){
	int i;
	scanf("%d", &N);
	for (i = 0; i < N; i++){
		scanf("%d", &P[i]);
	}
}

int solve(void){
	int i, sum = P[0];
	for(i = 1; i < N; i++){
		if(P[i - 1] < P[i]) sum = sum - P[i - 1] + P[i];
	}
	return sum;
}

int main(void){
	int ans = -1;
	Input_Data();
	ans = solve();
	printf("%d\n", ans);
	return 0;
}
